eth-contracts/contracts/service/ServiceProviderFactory.sol

```diff
@@ -272,4 +312,60 @@ contract ServiceProviderFactory is RegistryContract {
             _endpoint
         );
     }
+
+    function isValidServiceType(bytes32 _serviceType)
+    external view returns (bool isValid) 
+    {
+      for (uint i = 0; i < validServiceTypes.length; i ++) {
+        if (validServiceTypes[i] == _serviceType) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    function getValidServiceTypes()
+    external view returns (bytes32[] memory types) 
+    {
+      return validServiceTypes;
+    }
+
+    function getServiceStakeInfo(bytes32 _serviceType) 
+    external view returns (uint min, uint max)
+    {
+      return (serviceTypeStakeRequirements[_serviceType].minStake, serviceTypeStakeRequirements[_serviceType].maxStake);
+    }
+
+    function getAccountStakeBounds(address sp)
```

Comment:
// compute the min and max acceptable stake for service provider given all their registered services